File: 08-bubble-sort/src/BubbleSort.java
import java.util.Scanner;
import java.util.Arrays;

public class BubbleSort {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);

        System.out.println("Enter array size: ");
        int size = in.nextInt();
        int[] arr = new int[size];

        System.out.println("Enter elements: ");
        for (int i = 0; i < size; i++) {
            arr[i] = in.nextInt();
        }

        System.out.println("\n\nSorted array:");

        bubbleSort(arr);
        System.out.println(Arrays.toString(arr));

    }

    static void bubbleSort(int[] arr) {
        for (int i = 0; i < arr.length; i++) {
            int flag = 0;
            for (int j = 1; j < arr.length - i; j++) {
                if (arr[j] < arr[j - 1]) {
                    int temp = arr[j];
                    arr[j] = arr[j - 1];
                    arr[j - 1] = temp;

                    flag = 1;
                }

            }

            if (flag == 0) {
                break;
            }

        }

        // System.out.print("\n\nSorted array = ");
        // for (int i = 0; i < size; i++) {
        // System.out.print(arr[i] + " ");

        // }

    }

}

File: striver-sheet/bubble-sort.js
function bubbleSort(arr){
  for (let i = arr.length - 1; i >= 1; i--) {
    let isSwapped = false;
    for(let j = 0; j <= i - 1; j++){
      if(arr[j] > arr[j + 1]){
        let temp = arr[j + 1];
        arr[j + 1] = arr[j];
        arr[j] = temp;
        isSwapped = true;
      }
    }
    if(isSwapped === false){
      break;
    }
  }
  return arr;
}

let arr1 = [13, 46, 24, 52, 20, 9];
console.log(arr1);
console.log(bubbleSort(arr1));